static/js/robot.js

```diff
@@ -594,3 +618,90 @@ this.onmessage = function(e) {
+     var sensorsY = [this.robot.Y - 5*Math.cos(this.robot.dir/180*Math.PI), this.robot.Y, this.robot.Y + 5*Math.cos(this.robot.dir/180*Math.PI)]
+     var sensorsX = [this.robot.X - 5*Math.sin(-this.robot.dir/180*Math.PI), this.robot.X, this.robot.X + 5*Math.sin(-this.robot.dir/180*Math.PI)]
+
+     var tapeLines = this.robot.tapeLines
+     let total = []
+     let totalLine = 0
+     let i = 0
+     for (;i<3;i++){
+       let sensor_x = sensorsX[i]
+       let sensor_y = sensorsY[i]
+       // https://www.geeksforgeeks.org/program-for-point-of-intersection-of-two-lines/
+       let totalLine = 0
+       for (const tapeLine of tapeLines){
+         let m = tapeLine.slope
+         if (m === "horizontal") {
+           let distY = Math.abs(sensor_y-tapeLine.startY)
+           if (tapeLine.startX <= sensor_x && sensor_x <= tapeLine.endX) {
+             let distSquared = (distY*distY)
+             totalLine += Math.min(1,3/distSquared)
+           } else {
+             let distX = Math.min(Math.abs(tapeLine.startX-sensor_x), Math.abs(tapeLine.endX-sensor_x))
+             let distSquared = (distY*distY) + (distX*distX)
+             totalLine += Math.min(1,3/distSquared)
+             }
+         } else if (m === "vertical") {
+           let distX = Math.abs(sensor_x-tapeLine.startX)
+           if (tapeLine.startY <= sensor_y && sensor_y <= tapeLine.endY) {
+             let distSquared = (distX*distX)
+             totalLine += Math.min(1,3/distSquared)
+           } else {
+             let distY = Math.min(Math.abs(tapeLine.startY-sensor_y), Math.abs(tapeLine.endY-sensor_y))
+             let distSquared = (distY*distY) + (distX*distX)
+             totalLine += Math.min(1,3/distSquared)
+             }
+           } else {
+           // check if intersection point is inside the tapeLine
+             if ((sensor_x < tapeLine.startX && sensor_x < tapeLine.endX) || (sensor_x > tapeLine.startX && sensor_x > tapeLine.endX)){
+               let startDistX = Math.abs(sensor_x-tapeLine.startX)
+               let startDistY = Math.abs(sensor_y-tapeLine.startY)
+               let endDistX = Math.abs(sensor_x-tapeLine.endX)
+               let endDistY = Math.abs(sensor_y-tapeLine.startY)
+               let distSquared = Math.min((startDistX*startDistX+startDistY*startDistY),(endDistX*endDistX+endDistY*endDistY))
+               totalLine += Math.min(1,3/distSquared)
+           } else {
+             let m1 = -1/m
+             let c = tapeLine.y_int
+             let c1 = sensor_y-m1*sensor_x
+             let determinant = -m + m1
+             let x = (c - c1)/determinant
+             let y = (-m*c1 + m1*c)/determinant
+             let distX = Math.abs(sensor_x-x)
+             let distY = Math.abs(sensor_y-y)
+             let distSquared = (distX*distX)+(distY*distY)
+             totalLine += Math.min(1,3/distSquared)
+           }
+         }
+       }
+       total.push(Math.min(totalLine, 1))
+     }
+     console.log(total)
```

Comment:
We should delete this line. We can maybe add a Sensors tab (next to Code and Gamepad) that lists all current sensor readings. If we do this, we will actually have to read all the sensors every tick, nullifying my comment above. If this affects the performance of the simulator, we could look into doing something like updating the sensor readings every other tick, or every 5 ticks.